code-challenges/301/for-each/challenges-01.test.js

```diff
@@ -0,0 +1,195 @@
+CHALLENGE 1
+
+Write a function named greeting that takes in a string and returns the string in all uppercase letters.
+
+Then, write a function named speaker that takes in a string and a callback function. The speaker function should return the string in all uppercase letters only by invoking the callback.
+------------------------------------------------------------------------------------------------ */
+
+const greeting = (word) => word.toUpperCase();
+
+const speaker = (message, callback) => callback(message);
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 2
+
+Write a function named addValues that takes in an array and a value and pushes the value into the array. This function does not need a return statement.
+
+Then, write a function named addNumbers that takes in four arguments:
+  - A number to be added to an array
+  - An array into which the number should be added
+  - The number of times the number should be added
+  - A callback function to use to add the numbers to the array (Hint: you already defined it)
+
+Within the addNumbers function, invoke the callback function as many times as necessary, based on the third argument of the addNumbers function.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const addValues = (arr, value) => {
+  arr.push(value);
+}
+
+const addNumbers = (num, arr, times, callback) => {
+  for(var i = 0; i < times; i++) {
+    callback(arr,num);
+  }
+  return arr;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 3
+
+Write a function named removeOne that takes in a number and an array. If the number divided by three has a remainder of two, pop one element off of the array.
+
+Hint: you may want to look into the modulo operation.
+
+Then, write a function named removeElements that takes in an array and a callback. This function should use a for loop to iterate over the array and invoke the callback once for each element in the array.
+
+Return the modified array.
+------------------------------------------------------------------------------------------------ */
+
+const removeOne = (num, arr) => {
+  if ((num % 3) === 2 ) {
+    arr.pop();
+  }
+}
+
+const removeElements = (arr, callback) => {
+  for (let i = 0; i < arr.length; i++) {
+    callback(arr[i], arr);
+  }
+  return arr;
+}
+
+/* ------------------------------------------------------------------------------------------------
+CHALLENGE 4
+
+Write a function named removeWithForEach that produces the same output as challenge 3, but uses forEach.
+------------------------------------------------------------------------------------------------ */
+
+const removeWithForEach = (arr, callback) => {
+  arr.forEach(function(value, array){
+    removeOne(value,arr)

```

Comment:
The callback you define in your function call above should be what's put here instead of hard coding in 'removeOne'. removeOne is passes in place of the callback function during the test, so instead of removeOne(value, arr), you could have just written callback(value, arr). Still works, so not a big deal, you just want to avoid hard coding in function calls when possible. Great job though.